diff --git a/react/cache/provider.tsx b/react/cache/provider.tsx
index fa7fecbf1f5405f40779385adf06cd8b92695a29..1a81138006d75bb6c758171e036bb17b67e9f946 100644
--- a/react/cache/provider.tsx
+++ b/react/cache/provider.tsx
@@ -48,7 +48,7 @@ export type ConvexQueryCacheOptions = {
    *
    * @default 300000
    */
-  expiration?: number;
+  expiration?: number | undefined;
 
   /**
    * How many "extra" idle query subscriptions are allowed to remain
@@ -56,7 +56,7 @@ export type ConvexQueryCacheOptions = {
    *
    * @default Infinity
    */
-  maxIdleEntries?: number;
+  maxIdleEntries?: number | undefined;
 
   /**
    * A debug flag that will cause information about the query cache
@@ -64,7 +64,7 @@ export type ConvexQueryCacheOptions = {
    *
    * @default false
    */
-  debug?: boolean;
+  debug?: boolean | undefined;
 };
 
 /**
